### PR TITLE
pferd: init at 3.0.1

### PR DESCRIPTION
**Pull request by 0xbe7a**
###### Motivation for this change

A tool for downloading course-related files from ILIAS, a widely used e-learning platform at German universities.

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Added a release notes entry if the change is major or breaking
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/tools/misc/pferd/default.nix [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+    aiohttp
+    beautifulsoup4
+    rich
+    keyring
+    certifi
+  ];
+
+  meta = with lib; {
+    homepage = "https://github.com/Garmelon/PFERD";
+    description = "Programm zum Flotten, Einfachen Runterladen von Dateien";
```
Comment (fabaff) [inline pkgs/tools/misc/pferd/default.nix:33]:
```suggestion
    description = "Tool for downloading course-related files from ILIAS";
```

File: pkgs/tools/misc/pferd/default.nix [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+, fetchFromGitHub
+, aiohttp
+, beautifulsoup4
+, rich
+, keyring
+, certifi
+}:
+
+buildPythonApplication rec {
+  pname = "PFERD";
```
Comment (fabaff) [inline pkgs/tools/misc/pferd/default.nix:12]:
```suggestion
  pname = "pferd";
```

File: pkgs/tools/misc/pferd/default.nix (hunk 1 above)
Comment (legendofmiracles) [inline pkgs/tools/misc/pferd/default.nix:33]:
```suggestion
    description = "Pogram to easily download files of ILIAS";
```

**Review by legendofmiracles — COMMENTED** [review]
(ignore this comment - i can't remove it)

File: pkgs/tools/misc/pferd/default.nix [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+, buildPythonApplication
+, fetchFromGitHub
+, aiohttp
+, beautifulsoup4
+, rich
+, keyring
+, certifi
+}:
+
+buildPythonApplication rec {
```
Comment (fabaff) [inline pkgs/tools/misc/pferd/default.nix:11]: Please use `python3.pkgs.buildPythonApplication` and adjust the inputs. This allows to override dependencies.

```suggestion
python3.pkgs.buildPythonApplication rec {
```

File: pkgs/top-level/all-packages.nix [hunk 4]
```diff
@@ -860,6 +860,8 @@ in
 
   metapixel = callPackage ../tools/graphics/metapixel { };
 
+  pferd = python3Packages.callPackage ../tools/misc/pferd {};
```
Comment (fabaff) [inline pkgs/top-level/all-packages.nix:863]:
```suggestion
  pferd = callPackage ../tools/misc/pferd {};
```

File: pkgs/tools/misc/pferd/default.nix [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+  format = "pyproject";
+
+  src = fetchFromGitHub {
+    owner = "Garmelon";
+    repo = "PFERD";
+    rev = "v${version}";
+    sha256 = "1s0z8yajy3n194pnlqb48hy2n5qvhkzwbpksrdyds79vfq0b9rdl";
+  };
+
+  propagatedBuildInputs = [
```
Comment (fabaff) [inline pkgs/tools/misc/pferd/default.nix:23]:
```suggestion
  propagatedBuildInputs = with python3.pkgs; [
```

File: pkgs/tools/misc/pferd/default.nix [hunk 6]
```diff
@@ -0,0 +1,32 @@
...
+    rich
+    keyring
+    certifi
+  ];
+
+  meta = with lib; {
+    homepage = "https://github.com/Garmelon/PFERD";
+    description = "Tool for downloading course-related files from ILIAS";
+    license = licenses.mit;
+    maintainers = with lib.maintainers; [ _0xbe7a ];
```
Comment (sikmir) [inline pkgs/tools/misc/pferd/default.nix:30]:
```suggestion
    maintainers = with maintainers; [ _0xbe7a ];
```

Comment (0xbe7a) [reply pkgs/tools/misc/pferd/default.nix]: Thanks!

**Comment by fabaff** [comment]
Result of `nixpkgs-review pr 125210` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>pferd</li>
  </ul>
</details>

**Review by fabaff — APPROVED** [review]
- [x] package path fits guidelines
- [x] package name fits guidelines
- [x] package version fits guidelines
- [x] `meta.description` is set and fits guidelines
- [x] `meta.license` fits upstream license
- [x] `meta.maintainers` is set
- [x] package built (see https://github.com/NixOS/nixpkgs/pull/125210#issuecomment-853776991)
- [ ] all depending packages build
- [x] tested on x86_64

```bash
...ew/pr-125210]$ ./results/pferd/bin/pferd -h
usage: pferd [-h] [--version] [--config PATH] [--dump-config] [--dump-config-to PATH] [--debug-transforms] [--crawler NAME] [--working-dir PATH] [--explain] [--status] [--report] [--share-cookies] {local,kit-ilias-web} ...

optional arguments:
  -h, --help            show this help message and exit
  --version             show program's version number and exit
[...]
```

Looks good to me.

**Pull request merged**